src/files.ts

```diff
@@ -55,13 +55,13 @@ export function hasProject(): boolean {
 /**
  * Recursively finds all files that are part of the current project, and those that are ignored
  * by .claspignore and calls the passed callback function with the file lists.
- * @param {string} rootDir The project's root directory
+ * @param {string | undefined} rootDir The project's root directory
  * @param {FilesCallBack} callback The callback will be called with the following paramters
  *   error: Error if there's an error, otherwise null
  *   result: string[][], List of two lists of strings, ie. [nonIgnoredFilePaths,ignoredFilePaths]
  *   files?: Array<AppsScriptFile|undefined> Array of AppsScriptFile objects used by clasp push
  */
-export function getProjectFiles(rootDir: string, callback: FilesCallback): void {
+export function getProjectFiles(rootDir: string | undefined, callback: FilesCallback): void {
```

Comment:
`rootDir: string?`

Reply:
### Result of change

When I changed `rootDir: string?`, I have a error following to:
```
src/commands.ts:506:21 - error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string | null'.
  Type 'undefined' is not assignable to type 'string | null'.

506     getProjectFiles(rootDir, (err, projectFiles) => {
```

### Cause

I consider that the part of the cause is the following part.
https://github.com/google/clasp/blob/6384a2000675158f2e83e198dda1798fd5875abc/src/utils.ts#L45-L49

`rootDir?: string` is same to `rootDir: string | undefined` in property definition. But, it is same to `rootDir: string | null` in parameters definition.

### Solution

I considered that following to as solution:

1. rootDir parameter is changed required. If rootDir is falsy, Pass empty string to rootDir argument when to call getProjectFiles function.
2. No change source code.

Reply:
Sorry, that my mistake.

Can you try `rootDir?: string`?

If that doesn't work, then we can leave it as-is.

Reply:
I tried `rootDir?: string`. I have a error following to:
```
> tsc --project tsconfig.json && npm i -g --loglevel=error

src/files.ts:64:51 - error TS1016: A required parameter cannot follow an optional parameter.

64 export function getProjectFiles(rootDir?: string, callback: FilesCallback): void {
                                                     ~~~~~~~~
```

Reply:
Maybe 
```ts
export function getProjectFiles(rootDir: string = path.join('.', '/'), callback: FilesCallback): void {
```

And then for this line:
```ts
recursive(rootDir || path.join('.', '/'), (err, filePaths) => {
```
we can have
```ts
recursive(rootDir, (err, filePaths) => {
```

Reply:
I see, it went well.
Thank you.